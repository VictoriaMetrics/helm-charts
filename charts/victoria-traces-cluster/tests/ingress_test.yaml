suite: ingress templates test
chart:
  version: 0.1.1
  appVersion: 0.1.0
templates:
  - ingress.yaml
tests:
  - it: ingress should match snapshot
    set:
      vmauth:
        enabled: true
        ingress:
          enabled: true
          annotations:
            kubernetes.io/ingress.class: nginx
          hosts:
            - name: vtselect.node.local
              path: /select/test
              port: http
      vtinsert:
        ingress:
          enabled: true
          annotations: 
            kubernetes.io/ingress.class: alb
          tls:
            - secretName: insert-ingress-tls
              hosts:
                - insert.local
      vtselect:
        ingress:
          enabled: true
          annotations: 
            kubernetes.io/ingress.class: kong
    asserts:
      - matchSnapshot: {}
