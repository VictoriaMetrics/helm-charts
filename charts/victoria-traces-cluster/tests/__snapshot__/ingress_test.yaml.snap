ingress should match snapshot:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        kubernetes.io/ingress.class: nginx
      labels:
        app: vmauth
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: vt-cluster
        app.kubernetes.io/version: v1.116.0
        helm.sh/chart: victoria-traces-cluster-0.1.1
      name: RELEASE-NAME-vt-cluster-vmauth
      namespace: NAMESPACE
    spec:
      rules:
        - host: vtselect.node.local
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME-vt-cluster-vmauth
                    port:
                      name: http
                path: /select/test
                pathType: Prefix
  2: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        kubernetes.io/ingress.class: alb
      labels:
        app: vtinsert
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: vt-cluster
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: victoria-traces-cluster-0.1.1
      name: RELEASE-NAME-vt-cluster-vtinsert
      namespace: NAMESPACE
    spec:
      rules:
        - host: vtinsert.local
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME-vt-cluster-vtinsert
                    port:
                      name: http
                path: /insert
                pathType: Prefix
      tls:
        - hosts:
            - insert.local
          secretName: insert-ingress-tls
  3: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        kubernetes.io/ingress.class: kong
      labels:
        app: vtselect
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: vt-cluster
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: victoria-traces-cluster-0.1.1
      name: RELEASE-NAME-vt-cluster-vtselect
      namespace: NAMESPACE
    spec:
      rules:
        - host: vtselect.local
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME-vt-cluster-vtselect
                    port:
                      name: http
                path: /select
                pathType: Prefix
